src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpMetricSerializer.cs

```diff
@@ -0,0 +1,352 @@
+                    break;
+                }
+
+            case MetricType.Histogram:
+                {
+                    writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Metric_Data_Histogram, ProtobufWireType.LEN);
+                    int metricTypeLengthPosition = writePosition;
+                    writePosition += ReserveSizeForLength;
+
+                    // TODO: DO WORK HERE
+
+                    /*message Histogram {
+                          repeated HistogramDataPoint data_points = 1;
+                          AggregationTemporality aggregation_temporality = 2;
+                        }
+                    */
+
+                    ProtobufSerializer.WriteReservedLength(buffer, metricTypeLengthPosition, writePosition - (metricTypeLengthPosition + ReserveSizeForLength));
+                    break;
+                }
+
+            case MetricType.ExponentialHistogram:
+                {
+                    writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Metric_Data_Exponential_Histogram, ProtobufWireType.LEN);
+                    int metricTypeLengthPosition = writePosition;
+                    writePosition += ReserveSizeForLength;
+
+                    // TODO: DO WORK HERE
+
+                    /*message ExponentialHistogram {
+                          repeated ExponentialHistogramDataPoint data_points = 1;
+                          AggregationTemporality aggregation_temporality = 2;
+                        }
+                    */
+
+                    ProtobufSerializer.WriteReservedLength(buffer, metricTypeLengthPosition, writePosition - (metricTypeLengthPosition + ReserveSizeForLength));
+                    break;
+                }
+        }
+
+        ProtobufSerializer.WriteReservedLength(buffer, metricLengthPosition, writePosition - (metricLengthPosition + ReserveSizeForLength));
+        return writePosition;
+    }
+
+    private static int WriteNumberDataPoint<T>(byte[] buffer, int writePosition, int fieldNumber, in MetricPoint metricPoint, T value)
+    {
+        writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, fieldNumber, ProtobufWireType.LEN);
+        int dataPointLengthPosition = writePosition;
+        writePosition += ReserveSizeForLength;
+
+        if (typeof(T) == typeof(long))
+        {
+            // Casting to ulong is ok here as the bit representation for long versus ulong will be the same
+            // The difference would in the way the bit representation is interpreted on decoding side (signed versus unsigned)
+            writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.NumberDataPoint_Value_As_Int, (ulong)(long)(object)value!);
```

Comment:
why we need such a complex cast?

Reply:
this was copied verbatim from the Geneva Exporter. I can investigate if there's a simpler cast

Reply:
Seem my comment above about specialization done by the JIT. This complex cast will be optimized during that.

Reply:
this can be removed when switching to method overloads.